### PR TITLE
fix prod.webpack.config.js

### PR DESCRIPTION
**Pull request by ghost**
error in prod.webpack.config.js

### CODE REVIEW

File: 03 Misc/08 DefinePlugin advanced/prod.webpack.config.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 var ExtractTextPlugin = require('extract-text-webpack-plugin');
 var path = require('path');
+var webpack = require('webpack');
```
Comment (nasdan) [inline 03 Misc/08 DefinePlugin advanced/prod.webpack.config.js:3]: Nice catch, thank you!

File: 03 Misc/08 DefinePlugin advanced/prod.webpack.config.js [hunk 2]
```diff
@@ -68,4 +69,4 @@ module.exports = function () {
       })
     ],
   });
-}
+})();
```
Comment (nasdan) [inline 03 Misc/08 DefinePlugin advanced/prod.webpack.config.js:72]: Why do you auto-invoke this function?

Comment (ghost) [reply 03 Misc/08 DefinePlugin advanced/prod.webpack.config.js:72]: oh, just return an obj...

Comment (nasdan) [reply 03 Misc/08 DefinePlugin advanced/prod.webpack.config.js:72]: Nice, what do you think if we remove `function` stuff and export something like:

```
module.exports = webpackMerge(commonConfig, {
...
})
```

Comment (ghost) [reply 03 Misc/08 DefinePlugin advanced/prod.webpack.config.js:72]: just ok，i thought the previous is your code style😁

Comment (nasdan) [reply 03 Misc/08 DefinePlugin advanced/prod.webpack.config.js:72]: It was a mistake. Could you please update sample with this change to resolve this PR? Thank you so much for you contribution

**Pull request closed without merging**